go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go

```diff
@@ -186,64 +184,36 @@ type txThrottlerState struct {
 // This function calls tryCreateTxThrottler that does the actual creation work
 // and returns an error if one occurred.
 func NewTxThrottler(env tabletenv.Env, topoServer *topo.Server) TxThrottler {
-	txThrottler, err := tryCreateTxThrottler(env, topoServer)
-	if err != nil {
-		log.Errorf("Error creating transaction throttler. Transaction throttling will"+
-			" be disabled. Error: %v", err)
-		// newTxThrottler with disabled config never returns an error
-		txThrottler, _ = newTxThrottler(env, topoServer, &txThrottlerConfig{enabled: false})
-	} else {
-		log.Infof("Initialized transaction throttler with config: %+v", txThrottler.config)
-	}
-	return txThrottler
-}
-
-// InitDBConfig initializes the target parameters for the throttler.
-func (t *txThrottler) InitDBConfig(target *querypb.Target) {
-	t.target = proto.Clone(target).(*querypb.Target)
-}
-
-func tryCreateTxThrottler(env tabletenv.Env, topoServer *topo.Server) (*txThrottler, error) {
-	if !env.Config().EnableTxThrottler {
-		return newTxThrottler(env, topoServer, &txThrottlerConfig{enabled: false})
-	}
+	throttlerConfig := &txThrottlerConfig{enabled: false}
+
+	if env.Config().EnableTxThrottler {
+		// Clone tsv.TxThrottlerHealthCheckCells so that we don't assume tsv.TxThrottlerHealthCheckCells
+		// is immutable.
+		healthCheckCells := make([]string, len(env.Config().TxThrottlerHealthCheckCells))
+		copy(healthCheckCells, env.Config().TxThrottlerHealthCheckCells)
```

Comment:
My personal preference is to use a simpler one-liner:
```suggestion
      healthCheckCells := append(env.Config().TxThrottlerHealthCheckCells)
```

Reply:
@shlomi-noach SGTM thanks 👍

Reply:
@shlomi-noach looks like this fails the Static Code analysis CI with:

```bash
Error: go/vt/vttablet/tabletserver/txthrottler/tx_throttler.go:192:23: SA4021: x = append(y) is equivalent to x = y (staticcheck)
		healthCheckCells := append(env.Config().TxThrottlerHealthCheckCells)
```

So I'll go with what the linter is suggesting. I think defining a new var will create a copy too

Reply:
Huh, OK TIL.

> The underlying array is reused if there is enough capacity

I didn't think this through. And of course it makes perfect sense.